### PR TITLE
[snp-security]Remove vault component

### PR DESCRIPTION
**Pull request by tuteng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform-security/values.yaml [hunk 1]
```diff
@@ -1652,7 +1642,8 @@ prometheus:
     node_exporter: true
     proxy: true
     zookeeper: true
-    vault: true
+    # Deprecated
```
Comment (ericsyh) [inline charts/sn-platform-security/values.yaml:1645]: I think we can remove this directly.

File: charts/sn-platform-security/values.yaml [hunk 2]
```diff
@@ -372,8 +361,9 @@ auth:
     # streamnative-console
     streamnative-console: "super,pulsar-manager-admin"
   # Enable vault based authentication
+  # Deprecated
   vault:
-    enabled: true
+    enabled: false
```
Comment (ericsyh) [inline charts/sn-platform-security/values.yaml:366]: I think we can remove this directly.

**Review by ericsyh — APPROVED** [review]

File: charts/sn-platform-security/values.yaml (hunk 1 above)
Comment (tuteng) [reply charts/sn-platform-security/values.yaml:1639]: Fixed

File: charts/sn-platform-security/values.yaml (hunk 2 above)
Comment (tuteng) [reply charts/sn-platform-security/values.yaml]: Fixed

File: charts/sn-platform-security/values.yaml [hunk 3]
```diff
@@ -1652,7 +1636,8 @@ prometheus:
     node_exporter: true
     proxy: true
     zookeeper: true
-    vault: true
+    # Deprecated
+    vault: false
```
Comment (ericsyh) [inline charts/sn-platform-security/values.yaml:1640]: @tuteng did this config remove?

Comment (tuteng) [reply charts/sn-platform-security/values.yaml:1639]: Fixed

**Comment by tuteng** [comment]
@ericsyh PTAL

**Review by ericsyh — APPROVED** [review]

**Pull request merged**